addons/website_livechat/static/src/legacy/website_livechat_chatbot_test_script.js

```diff
@@ -84,7 +84,7 @@ publicWidget.registry.livechatChatbotTestScript = publicWidget.Widget.extend({
         utils.set_cookie('im_livechat_previous_operator_pid', '', -1);
 
         return this._super(...arguments).then(() => {
-            this.livechatButton = new LivechatButtonTestChatbot(this, this.$el.data());
+            this.livechatButton = new LivechatButtonTestChatbot(this, owl.Component.env.messaging, this.$el.data());
```

Comment:
`this.env` possible instead?

Reply:
This is a `web.public.widget`/`PublicWidget`, so they do not have `this.env`.

From what I see, this is a public widget globally registered, so I think `init(parent, ) {}` has parent, and `parent` is `RootWidget`, which itself is created with `createPublicRoot()`, which does deploy services, makeEnv()`... but it's not available from `RootWidget`...

So what I can do, is make `env` available to `RootWidget`.
Then for `LivechatButton`, receive `env` with `parent.env` and set it with `this.env`.

I would really avoid making all `PublicWidget` have `this.env`, because this would change all signatures and break all code, and I'm not confident in being able to make it work before v16!